expression/builtin_string.go

```diff
@@ -3372,6 +3372,11 @@ func (b *builtinFormatWithLocaleSig) evalString(row chunk.Row) (string, bool, er
 		b.ctx.GetSessionVars().StmtCtx.AppendWarning(errUnknownLocale.GenWithStackByArgs("NULL"))
 		locale = "en_US"
 	}
+	// TODO: support other locales.
+	if locale != "en_US" {
```

Comment:
I think there should be some normalization for case ( `strings.EqualFold(locale, "en_US")` etc) . In this branch:
```sql
mysql> SELECT FORMAT(12341.22,2, 'en_us');
+-----------------------------+
| FORMAT(12341.22,2, 'en_us') |
+-----------------------------+
| 12,341.22                   |
+-----------------------------+
1 row in set, 1 warning (0.00 sec)

mysql> show warnings;
+---------+------+-------------------------+
| Level   | Code | Message                 |
+---------+------+-------------------------+
| Warning | 1649 | Unknown locale: 'en_us' |
+---------+------+-------------------------+
1 row in set (0.00 sec)
```

Versus in MySQL it will match with lower case without a warning:
```sql
mysql [localhost:5731] {msandbox} ((none)) > SELECT FORMAT(12341.22,2, 'en_US');
+-----------------------------+
| FORMAT(12341.22,2, 'en_US') |
+-----------------------------+
| 12,341.22                   |
+-----------------------------+
1 row in set (0.00 sec)

mysql [localhost:5731] {msandbox} ((none)) > SELECT FORMAT(12341.22,2, 'en_AA');
+-----------------------------+
| FORMAT(12341.22,2, 'en_AA') |
+-----------------------------+
| 12,341.22                   |
+-----------------------------+
1 row in set, 1 warning (0.00 sec)

mysql [localhost:5731] {msandbox} ((none)) > SELECT FORMAT(12341.22,2, 'en_us');
+-----------------------------+
| FORMAT(12341.22,2, 'en_us') |
+-----------------------------+
| 12,341.22                   |
+-----------------------------+
1 row in set (0.00 sec)

```